google/cloud/aiplatform/models.py

```diff
@@ -215,10 +244,548 @@ def upload(
+                percentage of this Endpoint's traffic that
+                should be forwarded to that DeployedModel.
+                If a DeployedModel's ID is not listed in this
+                map, then it receives no traffic.
+
+                The traffic percentage values must add up to
+                100, or map must be empty if the Endpoint is to
+                not accept any traffic at a moment.
+            labels (Dict):
+                Optional. The labels with user-defined metadata to
+                organize your Endpoints.
+                Label keys and values can be no longer than 64
+                characters (Unicode codepoints), can only
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
+            project (str):
+                Optional. Project to retrieve endpoint from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve endpoint from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            endpoint (Endpoint):
+                Instantiated representation of the endpoint resource.
+        """
+
+        api_client = cls._instantiate_client(location=location, credentials=credentials)
+
+        create_endpoint_operation = cls._create(
+            api_client=api_client,
+            parent=initializer.global_config.common_location_path(
+                project=project, location=location
+            ),
+            display_name=display_name,
+            description=description,
+            traffic_split=traffic_split,
+            labels=labels,
+            metadata=metadata,
+        )
+
+        created_endpoint = None
+        endpoint_obj = cls(
+            endpoint=created_endpoint,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        create_endpoint_operation.add_update_resource_callback(
+            resource_noun_obj=endpoint_obj,
+            result_key="name",
+            api_get=lambda name: api_client.get_endpoint(name=name),
+        )
+
+        return endpoint_obj
+
+    @classmethod
+    def _create(
+        cls,
+        api_client: EndpointServiceClient,
+        parent: str,
+        display_name: str,
+        description: Optional[str] = None,
+        traffic_split: Optional[Dict] = {},
```

Comment:
is `traffic_split` needed when creating an endpoint?

also the default value should be `None`, not `{}`.

Reply:
traffic_split should be optional when creating an endpoint.

Reply:
The GAPIC create_endpoint method probably allows omitting the traffic_split field, and if the user passes in anything non empty the service will likely ignore it or raise an error. It should be just omitted in this method.